File: keeper/src/test/java/com/github/abhijitpparate/keeper/screen/note/presenter/NotePresenterTest.java
package com.github.abhijitpparate.keeper.screen.note.presenter;

import com.github.abhijitpparate.keeper.Constants;
import com.github.abhijitpparate.keeper.data.database.Note;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;

/**
 * Example local unit test, which will execute on the development machine (host).
 *
 * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
 */

@SuppressWarnings("FieldCanBeLocal")
@RunWith(MockitoJUnitRunner.class)
public class NotePresenterTest {

    // Presenter needs view and auth source
    // and provide a fake one as follows
    @Mock
    private NoteContract.View view;

    // Testing the presenter
    private NotePresenter presenter;

    @Before
    public void setUp(){
        MockitoAnnotations.initMocks(this);

        // Instantiate the presenter using the fake auth source and view
        presenter = new NotePresenter(view);
        presenter.subscribe();
    }

    @Test
    public void onSaveClick(){
        presenter.onSaveClick();
        verify(view).showHomeScreen();
    }

    @Test
    public void onLoadNote(){
        Note note = Constants.FAKE_NOTES.get(0);
        presenter.loadNote(note.getNoteId());
        verify(view).setNote(note);
        verify(view).setNoteTitle(note.getTitle());
        verify(view).setNoteBody(note.getBody());
    }

    @Test
    public void onChecklistClick(){
        presenter.onChecklistClick(true);
        verify(view).switchToChecklist();
    }

    @Test
    public void onLocationClick(){
        presenter.onLocationClick();
        verify(view).showPlacePicker();
    }



}

File: keeper/src/test/java/com/github/abhijitpparate/keeper/screen/note/presenter/NotePresenterTest.kt
package com.github.abhijitpparate.keeper.screen.note.presenter

import com.github.abhijitpparate.keeper.Constants

import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.MockitoAnnotations
import org.mockito.runners.MockitoJUnitRunner

import org.mockito.Mockito.verify

/**
 * Example local unit test, which will execute on the development machine (host).

 * @see [Testing documentation](http://d.android.com/tools/testing)
 */

@RunWith(MockitoJUnitRunner::class)
class NotePresenterTest {

    // Presenter needs view and auth source
    // and provide a fake one as follows
    @Mock
    private val view: NoteContract.View? = null

    // Testing the presenter
    private var presenter: NotePresenter? = null

    @Before
    fun setUp() {
        MockitoAnnotations.initMocks(this)

        // Instantiate the presenter using the fake auth source and view
        presenter = view?.let { NotePresenter(it) }
        presenter!!.subscribe()
    }

    @Test
    fun onSaveClick() {
        presenter!!.onSaveClick()
        verify<NoteContract.View>(view).showHomeScreen()
    }

    @Test
    fun onLoadNote() {
        val note = Constants.FAKE_NOTES[0]
        presenter!!.loadNote(note.noteId!!)
        verify<NoteContract.View>(view).setNote(note)
        verify<NoteContract.View>(view).noteTitle = note.title!!
        verify<NoteContract.View>(view).noteBody = note.body!!
    }

}